### ISSUE TITLE
r2.pred = 0 for almost all variants

### ISSUE THREAD

**Issue by mcarolborges** [zkutalik/ssimp_software#72]
Hi,

I have used Ssimp version 0.4 to impute several GWASes to 1000G (EUR population). For all GWASes, the resulting outputs have more than 90% of variants with r2.pred=0. Would you know what might be causing the problem?

Thank you.

Best wishes,
Carolina

**Comment by aaronmcdaid** [comment]
This is indeed strange that to how a low R2, when your GWAS file has so many SNPs in it. The 'lambda' value in the output file seems large. Any thoughts, @sinarueeger ? (Carolina has confirmed this is still an outstanding issue)

**Comment by aaronmcdaid** [comment]
In the last few minutes, I reran it with a lower lambda, 0.01 in instead of the default(*), I get much fewer instances of r2.pred=0. I think the lesson here is that the r2.pred will naturally be small or zero is the size of the reference panel is small

The default lambda is 2/sqrt(n), where `n` is the size of the reference panel. In the case of 1KG/EUR, that's 0.089=2/sqrt(503)

**Comment by mcarolborges** [comment]
Makes sense. Would you or @sinarueeger be able to advise what we should do in this case?
Many thanks!

**Comment by aaronmcdaid** [comment]
@sinarueeger  and I discussed that briefly, and I'm sure we'll discuss it in more detail again soon. The European subpopulation is relatively small, just 503 people, so the accuracy will always be low. The UK10K data could be used as a much larger reference dataset; if I remember correctly, it has approximately 2,000 people of European descent (but I'm not certain!)

In the past, we did discuss the possibility of automatically estimating the optimal lambda from the data. Sina and Zoltán have a method, but it's not in `ssimp` at the moment. This might improve the accuracy a little. But it has limitations - the size of the reference panel is always going to be a factor. A small reference panel, even with the best method and parameters, will always be limited.

When the r<sup>2</sup> is reported as zero, it actually means the r<sup>2</sup> was negative and we replace it with. We can see in equation 8 of our paper [1] that it can be negative if q<sub>eff</sub> is low. I'm sure there's more research to be done on maximizing the quality of the estimates, it's something I hope to return to someday!

[1] https://journals.plos.org/plosgenetics/article/comments?id=10.1371/journal.pgen.1007371 that
